bgpd/bgpd.c

```diff
@@ -4116,8 +4116,17 @@ int bgp_delete(struct bgp *bgp)
 	EVENT_OFF(bgp->t_update_delay);
 	EVENT_OFF(bgp->t_establish_wait);
 
+	if (bgp->inst_type == BGP_INSTANCE_TYPE_DEFAULT &&
+	    (bgp_table_top(bgp->rib[AFI_IP][SAFI_MPLS_VPN]) ||
+	     bgp_table_top(bgp->rib[AFI_IP6][SAFI_MPLS_VPN]))) {
+		if (BGP_DEBUG(zebra, ZEBRA))
+			zlog_debug("Marking the deleting default bgp instance as hidden");
+		SET_FLAG(bgp->flags, BGP_FLAG_INSTANCE_HIDDEN);
+	}
+
 	/* Set flag indicating bgp instance delete in progress */
-	SET_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS);
+	if (!IS_BGP_INSTANCE_HIDDEN(bgp))
```

Comment:
if a bgp instance is being deleted, and then restored the deletion flag should be UNSET there instead.